docs/source/format/CanonicalExtensions.rst

```diff
@@ -72,4 +72,57 @@ same rules as laid out above, and provide backwards compatibility guarantees.
 Official List
 =============
 
-No canonical extension types have been standardized yet.
+Fixed shape tensor
+==================
+
+* Extension name: `arrow.fixed_shape_tensor`.
+
+* The storage type of the extension: ``FixedSizeList`` where:
+
+  * **value_type** is the data type of individual tensors and
+    is an instance of ``pyarrow.DataType`` or ``pyarrow.Field``.
+  * **list_size** is the product of all the elements in tensor shape.
+
+* Extension type parameters:
+
+  * **value_type** = Arrow DataType of the tensor elements
+  * **shape** = shape of the contained tensors as a tuple
+
+* Description of the serialization:
+
+  The metadata must be a valid JSON object including shape of
+  the contained tensors as an array with key **"shape"**
+
+  - example: ``{ "shape": [2, 5]}``
+
+  and optional:
+
+  - **"dim_names"** holds explicit names to tensor dimensions
+    as an array. The length of it should be equal to the shape
+    length and equal to the number of dimensions.
+
+    Example of dim_names metadata for NCHW ordered data:
+
+    ``{ "shape": [100, 200, 500], "dim_names": ["C", "H", "W"]}``
+
+    The ``dim_names`` metadata can be added if the dimensions have
+    well-known names that can map to the physical order (row-major).
+
+  - **"permutation"** holds indexes of the desired ordering of the
+    original dimensions. Also defined as an array.
```

Comment:
Suggestion to add a more explicit explanation of what those values should be, something like: "The indices contain a permutation of the values [0, 1, .., N-1] where N is the number of dimensions. The permutation indicates which dimension of the logical order corresponds to which dimension of the physical tensor (the i-th dimension of the logical view corresponds to the dimension with number ``permutations[i]`` of the physical tensor).

Reply:
Perhaps something like:
```
  - **"permutation"** holds indexes of the desired logical ordering of the
    existing physical dimensions and can be used to derive strides.
    Also defined as an array.
```
Also adding an example of permutation for representing NHWC data as NCHW could be something to do? We should also state the convention somewhere if we're going to use it for examples (N = element, H = image height, W = image width, C = color channel).

Reply:
Sorry, I forgot to answer your comment here. I think the example of the permutation representing NHWC data as NCHW belongs to the implementations and the documentation of it. I plan to add examples of the usage and this would be a great thing to add.

Reply:
@rok I have added an example of permutation for representing NCHW data as NHWC in the Python bindings documentation PR: https://github.com/apache/arrow/pull/34957/files